spec/rspec/mocks/partial_double_spec.rb

```diff
@@ -383,6 +383,22 @@ def defined_private_method
         object.implemented
       end
 
+      it 'wraps proxies in a reentrant mutex so we dont trigger deadlocks' do
```

Comment:
We don't have any specs along these lines there, I felt it was similar in spirit to these specs and this is surfaced by this call back, if its moved elsewhere I'll need another way to trigger it.